external/storm-redis/src/main/java/org/apache/storm/redis/bolt/RedisFilterBolt.java

```diff
@@ -0,0 +1,146 @@
+ * <p/>
+ * Note2: If you want to just query about existence of key regardless of actual data type,
+ * specify STRING to data type of RedisFilterMapper.
+ */
+public class RedisFilterBolt extends AbstractRedisBolt {
+    private final RedisFilterMapper filterMapper;
+    private final RedisDataTypeDescription.RedisDataType dataType;
+    private final String additionalKey;
+
+    /**
+     * Constructor for single Redis environment (JedisPool)
+     * @param config configuration for initializing JedisPool
+     * @param filterMapper mapper containing which datatype, query key that Bolt uses
+     */
+    public RedisFilterBolt(JedisPoolConfig config, RedisFilterMapper filterMapper) {
+        super(config);
+
+        this.filterMapper = filterMapper;
+
+        RedisDataTypeDescription dataTypeDescription = filterMapper.getDataTypeDescription();
+        this.dataType = dataTypeDescription.getDataType();
+        this.additionalKey = dataTypeDescription.getAdditionalKey();
+    }
+
+    /**
+     * Constructor for Redis Cluster environment (JedisCluster)
+     * @param config configuration for initializing JedisCluster
+     * @param filterMapper mapper containing which datatype, query key that Bolt uses
+     */
+    public RedisFilterBolt(JedisClusterConfig config, RedisFilterMapper filterMapper) {
+        super(config);
+
+        this.filterMapper = filterMapper;
+
+        RedisDataTypeDescription dataTypeDescription = filterMapper.getDataTypeDescription();
+        this.dataType = dataTypeDescription.getDataType();
+        this.additionalKey = dataTypeDescription.getAdditionalKey();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void execute(Tuple input) {
+        String key = filterMapper.getKeyFromTuple(input);
+
+        boolean found;
+        JedisCommands jedisCommand = null;
+        try {
+            jedisCommand = getInstance();
+
+            switch (dataType) {
+                case STRING:
+                    found = jedisCommand.exists(key);
+                    break;
+
+                case SET:
+                    if (additionalKey == null) {
+                        throw new IllegalArgumentException("additionalKey should be defined");
+                    }
+                    found = jedisCommand.sismember(additionalKey, key);
+                    break;
+
+                case HASH:
+                    found = jedisCommand.hexists(additionalKey, key);
+                    break;
+
+                case SORTED_SET:
+                    found = jedisCommand.zrank(additionalKey, key) != null;
+                    break;
+
+                case HYPER_LOG_LOG:
+                    found = jedisCommand.pfcount(key) > 0;
+                    break;
+
+                case GEO:
+                    List<GeoCoordinate> geopos = jedisCommand.geopos(additionalKey, key);
+                    found = geopos != null && geopos.size() > 0;
```

Comment:
minor. Better to wrap in parentheses for clarity
` found = (geopos != null && geopos.size() > 0)`